pkg/atlantis/README.md

```diff
@@ -0,0 +1,165 @@
+# `atlantis` integration
+
+## Atlantis Repo Config Generation
+
+`atmos` supports generating [Repo Level atlantis.yaml Config](https://www.runatlantis.io/docs/repo-level-atlantis-yaml.html) for `atmos` components
+and stacks.
+
+The following `atmos` commands will first generate the varfiles for all components in all stacks,
+then generate the `atlantis.yaml` repo config file:
+
+```bash
+atmos terraform generate varfiles --file-template=varfiles/{tenant}-{environment}-{stage}-{component}.tfvars.json
+atmos atlantis generate repo-config --config-template config-1 --project-template project-1 --workflow-template workflow-1
+```
+
+__NOTE:__ All paths, `--file-template` in the `atmos terraform generate varfiles` command, and in the `atlantis` config in `atmos.yaml`,
+should be relative to the root of the repo. 
+
+Supported context tokens: `{namespace}`, `{tenant}`, `{environment}`, `{region}`, `{stage}`, `{component}`, `{component-path}`.
+
+You can run these commands manually and commit the generated varfiles and `atlantis.yaml` repo config.
+
+If you want to generate `atlantis.yaml` on the server dynamically,
+you can add the following run commands to [pre_workflow_hooks](https://www.runatlantis.io/docs/pre-workflow-hooks.html#pre-workflow-hooks).
+The `atlantis.yaml` repo config file will be generated right before Atlantis parses it.
+
+```yaml
+repos:
+  - id: /.*/
+    pre_workflow_hooks:
+      - run: |
+          atmos terraform generate varfiles --file-template=varfiles/{tenant}-{environment}-{stage}-{component}.tfvars.json
+          atmos atlantis generate repo-config --config-template config-1 --project-template project-1 --workflow-template workflow-1
+```
+
+Note that the `-file-template` parameter in the `atmos terraform generate varfiles` command must match the following two settings in `atmos.yaml`:
+
+- `when_modified` must use the same template with the context tokens - this will allow Atlantis to check if any of the generated variables were
+  modified
+- workflow `extra_args` must use the same template with the context tokens - this will allow Atlantis to run Terraform commands with the
+  correct `-var-file` parameters
+
+```yaml
+# atmos.yaml CLI config
+
+# Integrations
+integrations:
+
+  # Atlantis integration
+  # https://www.runatlantis.io/docs/repo-level-atlantis-yaml.html
+  atlantis:
+    # Path and name of the Atlantis config file `atlantis.yaml`
+    # Supports absolute and relative paths
+    # All the intermediate folders will be created automatically (e.g. `path: /config/atlantis/atlantis.yaml`)
+    # Can be overridden on the command line by using `--output-path` command-line argument in `atmos atlantis generate repo-config` command
+    # If not specified (set to an empty string/omitted here, and set to an empty string on the command line), the content of the file will be dumped to `stdout`
+    path: "atlantis.yaml"
```

Comment:
Can we have an option to override this with an env var or `--` switch ?

```
# for stdout
atmos atlantis generate repo-config --path="" # ...
# for atlantis.yaml
atmos atlantis generate repo-config --path="atlantis.yaml" # ...
```

Reply:
Can you document that `--output-path=/dev/stdout` flag? That would be helpful for people who want to output to stdout on the fly without having to modify the `atlantis.yaml` when testing.